stl/inc/flat_map

```diff
@@ -862,27 +864,30 @@ private:
         _Clear_flat_map_scope_guard _Guard{this};
 
         // Insert the new elements at the end
-        size_type _OldSize = size();
-        size_type _NewSize = _STD distance(_First, _Last);
+        const size_type _Old_size = size();
+        if constexpr (_Is_cpp17_fwd_iter_v<_InputIterator>) {
+            const auto _New_size = static_cast<size_type>(_STD distance(_First, _Last));
 
-        _Data.keys.reserve(_Data.keys.size() + _NewSize);
-        _Data.values.reserve(_Data.values.size() + _NewSize);
+            _Data.keys.reserve(_Data.keys.size() + _New_size);
+            _Data.values.reserve(_Data.values.size() + _New_size);

```

Comment:
Pre-existing: This is quadratic complexity. `vector::reserve()` is allowed to give you exactly as much memory as you request (and our implementation does), so if you attempt to help the library by calling `v.reserve(v.size() + N)` before inserting N additional elements, doing that in a loop will be quadratic complexity.

If we can't range-insert here, we probably shouldn't try to be clever with `reserve()`.